receiver/prometheusreceiver/targetallocator/manager_test.go

```diff
@@ -714,7 +714,8 @@ func TestTargetAllocatorJobRetrieval(t *testing.T) {
 							"__meta_prometheus_job": "node",
 						},
 						MetricRelabelConfig: &expectedMetricRelabelConfigTestResult{
-							JobName:            "job1",
+							JobName: "job1",
+							// XXXXXXXXXXXXXXXx bad
 							MetricRelabelRegex: relabel.MustNewRegexp("regex2"),
```

Comment:
the test is returning regex1, indicating that the relabel config is not getting updated properly.  I cannot figure out how this change would have done that.  race condition?

Reply:
Yes I think this is a race condition.  I think the test is being run before the second scrapeconfig call is made.

Reply:
The allocator waitgroup returns after only one response has been given, which is weirdly by design.

Reply:
The problem is that we call Refresh and then immediately check for the result, but the updated config takes a while to be set.  So we need to find a way to wait until we know the update has been done.